imageregistry/v1/types.go

```diff
@@ -306,6 +306,59 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+type EndpointAccessibility string
+
+var (
+	InternalEndpoint = EndpointAccessibility("Internal")
+	PublicEndpoint   = EndpointAccessibility("Public")
+)
+
+type Encryption string
+
+var (
+	ClearText = Encryption("ClearText")
+	AES256    = Encryption("AES256")
+	KMS       = Encryption("KMS")
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	EncryptionType Encryption `json:"encryptionType"`
+
+	KMSEncryptionAlibaba *KMSEncryptionAlibaba `json:"kms"`
+}
+
+type KMSEncryptionAlibaba struct {
+	KeyID string `json:"keyID"`
```

Comment:
doc this field (key id of what/from where?)

Reply:
is this security sensitive?